infrastructure/src/test/java/mops/controllers/FeedbackControllerTest.java

```diff
@@ -44,4 +48,18 @@ public void publicUserCantAccessFeedback() throws Exception {
         .andExpect(status().is2xxSuccessful())
         .andExpect(view().name("index"));
   }
+
+  @Test
+  @DisplayName("Public User Should Be Redirected To Login Page When Attempting To Go To Dozenten")
+  public void loginRedirectionForPublicUsersToDozentPage() throws Exception {
+    mvc.perform(get("/feedback/dozenten")).andExpect(status().is3xxRedirection())
+        .andExpect(((redirectedUrl("/sso/login"))));
+  }
+
+  @Test
+  @DisplayName("Public User Should Be Redirected To Login Page When Attempting To Go To Studenten")
+  public void loginRedirectionForPublicUsersToStudentPage() throws Exception {
+    mvc.perform(get("/feedback/studenten")).andExpect(status().is3xxRedirection())
+        .andExpect(((redirectedUrl("/sso/login"))));
+  }
```

Comment:
Isn't this a Keycloak Feature? Why do we need a Test for this?

Reply:
We're supposed to test all the scenarios and this is a valid scenario to test.

Reply:
We are NOT supposed to test Library Functions. Especially not with Resource intensive mvc tests that substantially increase test time. Since we have no control over where the redirect goes, you are really only testing, whether or not the Controller Method has the Roles Allowed annotation. Which you can do way more efficiently with an Arch Unit Test.